src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -275,7 +275,9 @@ def forward(self, x, seq_len=None):
             emb = torch.cat((freqs, freqs), dim=-1).to(x.device)
             self.cos_cached = emb.cos()[None, None, :, :]
             self.sin_cached = emb.sin()[None, None, :, :]
-        return self.cos_cached[:seq_len, ...].to(x.device), self.sin_cached[:seq_len, ...].to(x.device)
+        cos = self.cos_cached[:seq_len, ...].to(x.device).to(x.dtype)
+        sin = self.sin_cached[:seq_len, ...].to(x.device).to(x.dtype)
+        return cos, sin
```

Comment:
I think we might also want to control the type when creating the weights e.g. [like here for Llama](https://github.com/huggingface/transformers/blob/a04ebc8b33314c42349c3e12885960a292c9c9dd/src/transformers/models/llama/modeling_llama.py#LL103C1-L103C1)

cc @younesbelkada who know's more about this